### PR TITLE
Add image tagging using the MagicMirror releases

### LINKED ISSUE #36: Docker tags for the images
Can you please tag major versions of the image, so we can better control when we pull a new version of the image with breaking changes.

### PR DESCRIPTION
**Pull request by bastilimbach**
With this PR we now have 3 separate tags:
- `latest`: Is build & pushed only if the latest release is not yet on Docker Hub.
- `develop`: Is build & pushed daily and uses the `develop` branch of MagicMirror.
- `vX.Y.Z`: Is build & pushed only if a new release (tag) is published by MagicMirror and it is not yet on Docker Hub. 

Closes #36

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -14,13 +14,23 @@ before_script:
...
       echo "$DOCKER_PASS" | docker login -u ${DOCKER_USER} --password-stdin
       dopush="--push"
     fi
-  - docker buildx build --progress plain --platform=linux/amd64,linux/arm64,linux/arm/v7 ${dopush} -t ${DOCKER_USER}/docker-magicmirror:latest .
+  # Build the latest stable MagicMirror release (https://stackoverflow.com/a/51761312/4934537)
+  - >
+    latest_release=$(git ls-remote --tags --refs --sort="v:refname" https://github.com/MichMich/MagicMirror.git | tail -n1 | sed 's/.*\///')
+    if [ "$(docker manifest inspect bastilimbach/docker-magicmirror:"${latest_release}" > /dev/null; echo $?)" != 0 ]; then
+      docker buildx build --progress plain --platform=linux/amd64,linux/arm64,linux/arm/v7 "${dopush}" --build-arg branch="${latest_release}" -t "${DOCKER_USER}"/docker-magicmirror:"${latest_release}" .
+      docker buildx build --progress plain --platform=linux/amd64,linux/arm64,linux/arm/v7 "${dopush}" --build-arg branch="${latest_release}" -t "${DOCKER_USER}"/docker-magicmirror:latest .
```
Comment (Legion2) [inline .travis.yml]: you can build the image once and then tag the image with many tags

File: .travis.yml [hunk 2]
```diff
@@ -14,13 +14,23 @@ before_script:
...
+  # Login to Docker Hub and configure push behaviour if we are on master
   - >
     if [ "$TRAVIS_BRANCH" == "master" ] && [ "$TRAVIS_PULL_REQUEST" == "false" ]; then
       echo "$DOCKER_PASS" | docker login -u ${DOCKER_USER} --password-stdin
       dopush="--push"
     fi
-  - docker buildx build --progress plain --platform=linux/amd64,linux/arm64,linux/arm/v7 ${dopush} -t ${DOCKER_USER}/docker-magicmirror:latest .
+  # Build the latest stable MagicMirror release (https://stackoverflow.com/a/51761312/4934537)
+  - >
+    latest_release=$(git ls-remote --tags --refs --sort="v:refname" https://github.com/MichMich/MagicMirror.git | tail -n1 | sed 's/.*\///')
```
Comment (khassel) [inline .travis.yml]: I think there is a `\n` in `latest_release` which causes the problem (had something similar yesterday which costs me 3 hours ...), try e.g.

````
git ls-remote --tags --refs --sort="v:refname" https://github.com/MichMich/MagicMirror.git | tail -n1 | sed 's/.*\///' | tr '\n' ' '
````

File: .travis.yml (hunk 1 above)
Comment (bastilimbach) [reply .travis.yml]: Oh, you are right. Good point, thanks!

File: .travis.yml (hunk 2 above)
Comment (bastilimbach) [reply .travis.yml]: Oh please let this be it! I'm already investigating over an hour...
I'll try this immediately!

Comment (bastilimbach) [reply .travis.yml]: This didn’t fix the issue 😭

Comment (khassel) [reply .travis.yml]: no :(

may like this. If it does'nt work we may get a better error message ...

````
  - latest_release=$(git ls-remote --tags --refs --sort="v:refname" https://github.com/MichMich/MagicMirror.git | tail -n1 | sed 's/.*\///' | tr '\n' ' ')
  - >
    if [ "$(docker manifest inspect bastilimbach/docker-magicmirror:"${latest_release}" > /dev/null; echo $?)" != 0 ]; then
      docker buildx build --progress plain --platform=linux/amd64,linux/arm64,linux/arm/v7 "${dopush}" --build-arg branch="${latest_release}" -t "${DOCKER_USER}"/docker-magicmirror:"${latest_release}" -t "${DOCKER_USER}"/docker-magicmirror:latest .
    fi
````

**Comment by Legion2** [comment]
Try to define all the scripts in a bash file

**Comment by bastilimbach** [comment]
Yeah this might be an option. Will try it later 👍

File: .travis.yml (hunk 2 above)
Comment (bastilimbach) [reply .travis.yml]: Ok lets try it... If this doesn't print any useful error message I will try a separate bash file.

Comment (bastilimbach) [reply .travis.yml]: Ok we really got a better error message:
```
0.42s$ latest_release=$(git ls-remote --tags --refs --sort="v:refname" https://github.com/MichMich/MagicMirror.git | tail -n1 | sed 's/.*\///' | tr '\n' ' ')
The command "latest_release=$(git ls-remote --tags --refs --sort="v:refname" https://github.com/MichMich/MagicMirror.git | tail -n1 | sed 's/.*\///' | tr '\n' ' ')" exited with 0.
0.10s$ if [ "$(docker manifest inspect bastilimbach/docker-magicmirror:"${latest_release}" > /dev/null; echo $?)" != 0 ]; then
  docker buildx build --progress plain --platform=linux/amd64,linux/arm64,linux/arm/v7 "${dopush}" --build-arg branch="${latest_release}" -t "${DOCKER_USER}"/docker-magicmirror:"${latest_release}" -t "${DOCKER_USER}"/docker-magicmirror:latest .
fi
invalid reference format
"docker buildx build" requires exactly 1 argument.
See 'docker buildx build --help'.
Usage:  docker buildx build [OPTIONS] PATH | URL | -
Start a build
The command "if [ "$(docker manifest inspect bastilimbach/docker-magicmirror:"${latest_release}" > /dev/null; echo $?)" != 0 ]; then
  docker buildx build --progress plain --platform=linux/amd64,linux/arm64,linux/arm/v7 "${dopush}" --build-arg branch="${latest_release}" -t "${DOCKER_USER}"/docker-magicmirror:"${latest_release}" -t "${DOCKER_USER}"/docker-magicmirror:latest .
fi
" exited with 1.
```

Comment (bastilimbach) [reply .travis.yml]: Probably because "latest_release" is not available in the next command.

**Comment by Legion2** [comment]
there is a white space in the tag name 
`invalid tag "bastilimbach/docker-magicmirror:v2.11.0 ": invalid reference format`

**Comment by bastilimbach** [comment]
Yes this should be fixed now...
The empty space at the end was the issue.

**Comment by bastilimbach** [comment]
Holy... It seems to actually work.

**Comment by khassel** [comment]
schwere Geburt 😜

**Comment by bastilimbach** [comment]
Aber hallo -.-

**Pull request merged**